File: src/main/java/com/hospital/management/medicalrecord/service/MedicalRecordService.java
package com.hospital.management.medicalrecord.service;

import com.hospital.management.medicalrecord.dto.MedicalRecordRequestDTO;
import com.hospital.management.medicalrecord.dto.MedicalRecordResponseDTO;

import java.util.List;

public interface MedicalRecordService {

    MedicalRecordResponseDTO saveMedicalRecord(MedicalRecordRequestDTO dto);

    List<MedicalRecordResponseDTO> getAllMedicalRecords();

    MedicalRecordResponseDTO getMedicalRecordById(Long id);

    MedicalRecordResponseDTO updateMedicalRecord(Long id, MedicalRecordRequestDTO dto);

    void deleteMedicalRecord(Long id);
}

File: src/services/medicalRecordService.js
import api from "./api";

export const getAllMedicalRecords = () => {
  return api.get("/api/medical-records");
};

export const saveMedicalRecord = (record) => {
  return api.post("/api/medical-records", record);
};

export const deleteMedicalRecord = (id) => {
  return api.delete(`/api/medical-records/${id}`);
};
export const updateMedicalRecord = (id, record) => {
  return api.put(`/api/medical-records/${id}`, record);
};